pl_bolts/models/self_supervised/fixmatch/fixmatch_module.py

```diff
@@ -0,0 +1,196 @@
+
+        loss_u = (self.criteria_u(logits_u_strong, label_u_guess) * mask).mean()
+
+        loss = loss_x + self.hparams.coefficient_u * loss_u
+        self.log("loss", loss, on_step=True, on_epoch=True, logger=True)
+        self.log("loss_x", loss_x, on_step=True, on_epoch=True, logger=True)
+        self.log("loss_u", loss_u, on_step=True, on_epoch=True)
+        corr_u_label = (label_u_guess == label_u).float() * mask
+        self.log("num of acc@unlabeled", corr_u_label.sum().item(), on_step=True, on_epoch=True)
+        self.log("num of strong aug", mask.sum().item(), on_step=True, on_epoch=True)
+        self.log("num of mask", mask.mean().item(), on_step=True, on_epoch=True)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        images, labels = batch
+        logits = self.model(images)
+        loss = self.criteria_x(logits, labels)
+        acc1, acc5 = self.__accuracy(logits, labels, topk=(1, 5))
+        # ema eval
+        if self.ema_eval:
+            with torch.no_grad():
+                ema_model_update(self.model, self.ema_model, self.hparams.ema_m)
+        self.log("val_loss", loss, on_step=True, on_epoch=True)
+        self.log("val_acc1", acc1, on_step=True, prog_bar=True, on_epoch=True)
+        self.log("val_acc5", acc5, on_step=True, on_epoch=True)
+        return loss
+
+    def configure_optimizers(self):
+        optimizer = optim.SGD(
+            self.parameters(),
+            lr=self.hparams.lr,
+            momentum=self.hparams.momentum,
+            weight_decay=self.hparams.weight_decay,
+            nesterov=True,
+        )
+        scheduler = WarmupCosineLrScheduler(optimizer, max_iter=self.total_steps, warmup_iter=0)
+        return [optimizer], [scheduler]
+
+    def get_unlabled_logits_weak_probs(self, logits_u_weak):
+        probs = torch.softmax(logits_u_weak, dim=1)
+        if self.hparams.distribution_alignment:
+            self.prob_list.append(probs.mean(0))
+            if len(self.prob_list) > 32:
+                self.prob_list.pop(0)
+            prob_avg = torch.stack(self.prob_list, dim=0).mean(0)
+            probs = probs / prob_avg
+            probs = probs / probs.sum(dim=1, keepdim=True)
+        return probs
+
+    @staticmethod
+    def __accuracy(output, target, topk=(1,)):
+        """Computes the accuracy over the k top predictions for the specified values of k."""
+        with torch.no_grad():
+            maxk = max(topk)
+            batch_size = target.size(0)
+
+            _, pred = output.topk(maxk, 1, True, True)
+            pred = pred.t()
+            correct = pred.eq(target.view(1, -1).expand_as(pred))
+
+            res = []
+            for k in topk:
+                correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)
+                res.append(correct_k.mul_(100.0 / batch_size))
+            return res
+
+    @staticmethod
+    def add_model_specific_args(parent_parser):  # pragma: no-cover
```

Comment:
lets use `LightningCLI`

Reply:
Okay, I would try this.

I have just finished CoMatch updates.